### ISSUE TITLE
fcos使用voc数据集 map特别低

### ISSUE THREAD

**Issue by sharp-xiang** [open-mmlab/mmdetection#7736]
最后结果只有0.103

**Comment by sharp-xiang** [comment]
只有 0.103

**Comment by chhluo** [comment]
Please provide the config file you used and any modification to your code or dataset. Maybe there are mistakes in your operations.

**Comment by sharp-xiang** [comment]
dataset_type = 'VOCDataset'
data_root = 'data/VOCdevkit/'
img_norm_cfg = dict(
    mean=[102.9801, 115.9465, 122.7717], std=[1.0, 1.0, 1.0], to_rgb=False)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(
        type='Normalize',
        mean=[102.9801, 115.9465, 122.7717],
        std=[1.0, 1.0, 1.0],
        to_rgb=False),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1333, 800),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[102.9801, 115.9465, 122.7717],
                std=[1.0, 1.0, 1.0],
                to_rgb=False),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type='RepeatDataset',
        times=3,
        dataset=dict(
            type='VOCDataset',
            ann_file=['data/VOCdevkit/VOC2012/ImageSets/Main/trainval.txt'],
            img_prefix=['data/VOCdevkit/VOC2012/'],
            pipeline=[
                dict(type='LoadImageFromFile'),
                dict(type='LoadAnnotations', with_bbox=True),
                dict(type='Resize', img_scale=(1000, 600), keep_ratio=True),
                dict(type='RandomFlip', flip_ratio=0.5),
                dict(
                    type='Normalize',
                    mean=[123.675, 116.28, 103.53],
                    std=[58.395, 57.12, 57.375],
                    to_rgb=True),
                dict(type='Pad', size_divisor=32),
                dict(type='DefaultFormatBundle'),
                dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
            ])),
    val=dict(
        type='VOCDataset',
        ann_file='data/VOCdevkit/VOC2007/ImageSets/Main/test.txt',
        img_prefix='data/VOCdevkit/VOC2007/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[102.9801, 115.9465, 122.7717],
                        std=[1.0, 1.0, 1.0],
                        to_rgb=False),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='VOCDataset',
        ann_file='data/VOCdevkit/VOC2007/ImageSets/Main/test.txt',
        img_prefix='data/VOCdevkit/VOC2007/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 800),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[102.9801, 115.9465, 122.7717],
                        std=[1.0, 1.0, 1.0],
                        to_rgb=False),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
evaluation = dict(interval=1, metric='mAP')
optimizer = dict(
    type='SGD',
    lr=0.01,
    momentum=0.9,
    weight_decay=0.0001,
    paramwise_cfg=dict(bias_lr_mult=2.0, bias_decay_mult=0.0))
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
lr_config = dict(
    policy='step',
    warmup='constant',
    warmup_iters=500,
    warmup_ratio=0.3333333333333333,
    step=[8, 11])
runner = dict(type='EpochBasedRunner', max_epochs=12)
checkpoint_config = dict(interval=1)
log_config = dict(interval=10, hooks=[dict(type='TextLoggerHook')])
custom_hooks = [dict(type='NumClassCheckHook')]
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
opencv_num_threads = 0
mp_start_method = 'fork'
model = dict(
    type='FCOS',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=False),
        norm_eval=True,
        style='caffe',
        init_cfg=dict(
            type='Pretrained',
            checkpoint='open-mmlab://detectron/resnet50_caffe')),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        add_extra_convs='on_output',
        num_outs=5,
        relu_before_extra_convs=True),
    bbox_head=dict(
        type='FCOSHead',
        num_classes=20,
        in_channels=256,
        stacked_convs=4,
        feat_channels=256,
        strides=[8, 16, 32, 64, 128],
        loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(type='IoULoss', loss_weight=1.0),
        loss_centerness=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0)),
    train_cfg=dict(
        assigner=dict(
            type='MaxIoUAssigner',
            pos_iou_thr=0.5,
            neg_iou_thr=0.4,
            min_pos_iou=0,
            ignore_iof_thr=-1),
        allowed_border=-1,
        pos_weight=-1,
        debug=False),
    test_cfg=dict(
        nms_pre=1000,
        min_bbox_size=0,
        score_thr=0.05,
        nms=dict(type='nms', iou_threshold=0.5),
        max_per_img=100))
work_dir = 'fcos_test'
auto_resume = False
gpu_ids = range(0, 1)

**Comment by sharp-xiang** [comment]
学习率 resnet50 从'open-mmlab://detectron/resnet50_caffe改成torchvision.resnet也改过  都没啥用

**Comment by chhluo** [comment]
```
data = dict(
samples_per_gpu=2,
workers_per_gpu=2,
train=dict(
type='RepeatDataset',
times=3,
dataset=dict(
type='VOCDataset',
ann_file=['data/VOCdevkit/VOC2012/ImageSets/Main/trainval.txt'],
img_prefix=['data/VOCdevkit/VOC2012/'],
pipeline=[
dict(type='LoadImageFromFile'),
dict(type='LoadAnnotations', with_bbox=True),
dict(type='Resize', img_scale=(1000, 600), keep_ratio=True),
dict(type='RandomFlip', flip_ratio=0.5),
dict(
type='Normalize',
mean=[123.675, 116.28, 103.53],
std=[58.395, 57.12, 57.375],
to_rgb=True),
dict(type='Pad', size_divisor=32),
dict(type='DefaultFormatBundle'),
dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
])),
val=dict(
type='VOCDataset',
ann_file='data/VOCdevkit/VOC2007/ImageSets/Main/test.txt',
img_prefix='data/VOCdevkit/VOC2007/',
pipeline=[
dict(type='LoadImageFromFile'),
dict(
type='MultiScaleFlipAug',
img_scale=(1333, 800),
flip=False,
transforms=[
dict(type='Resize', keep_ratio=True),
dict(type='RandomFlip'),
dict(
type='Normalize',
mean=[102.9801, 115.9465, 122.7717],
std=[1.0, 1.0, 1.0],
to_rgb=False),
dict(type='Pad', size_divisor=32),
dict(type='ImageToTensor', keys=['img']),
dict(type='Collect', keys=['img'])
])
]),
test=dict(
type='VOCDataset',
ann_file='data/VOCdevkit/VOC2007/ImageSets/Main/test.txt',
img_prefix='data/VOCdevkit/VOC2007/',
pipeline=[
dict(type='LoadImageFromFile'),
dict(
type='MultiScaleFlipAug',
img_scale=(1333, 800),
flip=False,
transforms=[
dict(type='Resize', keep_ratio=True),
dict(type='RandomFlip'),
dict(
type='Normalize',
mean=[102.9801, 115.9465, 122.7717],
std=[1.0, 1.0, 1.0],
to_rgb=False),
dict(type='Pad', size_divisor=32),
dict(type='ImageToTensor', keys=['img']),
dict(type='Collect', keys=['img'])
])
]))
```

image mean and std are different for train pipeline and val pipeline . Please set the mean and std in val pipeline to the value used in train pipeline.

**Comment by sharp-xiang** [comment]
> ```
> data = dict(
> samples_per_gpu=2,
> workers_per_gpu=2,
> train=dict(
> type='RepeatDataset',
> times=3,
> dataset=dict(
> type='VOCDataset',
> ann_file=['data/VOCdevkit/VOC2012/ImageSets/Main/trainval.txt'],
> img_prefix=['data/VOCdevkit/VOC2012/'],
> pipeline=[
> dict(type='LoadImageFromFile'),
> dict(type='LoadAnnotations', with_bbox=True),
> dict(type='Resize', img_scale=(1000, 600), keep_ratio=True),
> dict(type='RandomFlip', flip_ratio=0.5),
> dict(
> type='Normalize',
> mean=[123.675, 116.28, 103.53],
> std=[58.395, 57.12, 57.375],
> to_rgb=True),
> dict(type='Pad', size_divisor=32),
> dict(type='DefaultFormatBundle'),
> dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
> ])),
> val=dict(
> type='VOCDataset',
> ann_file='data/VOCdevkit/VOC2007/ImageSets/Main/test.txt',
> img_prefix='data/VOCdevkit/VOC2007/',
> pipeline=[
> dict(type='LoadImageFromFile'),
> dict(
> type='MultiScaleFlipAug',
> img_scale=(1333, 800),
> flip=False,
> transforms=[
> dict(type='Resize', keep_ratio=True),
> dict(type='RandomFlip'),
> dict(
> type='Normalize',
> mean=[123.675, 116.28, 103.53],
> std=[58.395, 57.12, 57.375],
> to_rgb=False),
> dict(type='Pad', size_divisor=32),
> dict(type='ImageToTensor', keys=['img']),
> dict(type='Collect', keys=['img'])
> ])
> ]),
> test=dict(
> type='VOCDataset',
> ann_file='data/VOCdevkit/VOC2007/ImageSets/Main/test.txt',
> img_prefix='data/VOCdevkit/VOC2007/',
> pipeline=[
> dict(type='LoadImageFromFile'),
> dict(
> type='MultiScaleFlipAug',
> img_scale=(1333, 800),
> flip=False,
> transforms=[
> dict(type='Resize', keep_ratio=True),
> dict(type='RandomFlip'),
> dict(
> type='Normalize',
> mean=[123.675, 116.28, 103.53],
> std=[58.395, 57.12, 57.375],
> to_rgb=False),
> dict(type='Pad', size_divisor=32),
> dict(type='ImageToTensor', keys=['img']),
> dict(type='Collect', keys=['img'])
> ])
> ]))
> ```
> 
> train pipeline 和 val pipeline 的 image mean 和 std 是不同的。请将 val 管道中的平均值和标准设置为 train 管道中使用的值。无需再培训。修改后进行测试`CUDA_VISIBLE_DEVICES=0 python tools/test.py path_to_config.py path_to_checkpoint.pth --eval bbox`。

the problem has solved，thanks very much 谢谢大佬 困扰了好几天了
